src/Spago/Build.hs

```diff
@@ -87,10 +89,26 @@ build BuildOptions{..} maybePostBuild = do
 repl :: Spago m => [Purs.SourcePath] -> [Purs.ExtraArg] -> m ()
 repl sourcePaths passthroughArgs = do
   echoDebug "Running `spago repl`"
-  config <- Config.ensureConfig
-  deps <- Packages.getProjectDeps config
-  let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
-  Purs.repl globs passthroughArgs
+
+  hasPackagesDhall <- testfile "packages.dhall"
+  if hasPackagesDhall
+    then do
+      config <- Config.ensureConfig
+      deps <- Packages.getProjectDeps config
+      let globs = Packages.getGlobs deps <> Config.configSourcePaths config <> sourcePaths
+      Purs.repl globs passthroughArgs
+    else Temp.withSystemTempDirectory "spago-repl-tmp" $ \dir -> do
```

Comment:
In the past we had permissions issues with `withSystemTempDirectory`, so maybe we could use `withTempDirectory` with the [global cache directory](https://github.com/spacchetti/spago/blob/92e9c64bdde0373a64514def96b7f78738601909/src/Spago/GlobalCache.hs#L150) as base path?

Reply:
Ok, good to know!